### ISSUE TITLE
Bug in filter2D or error in doc

### ISSUE THREAD

**Issue by AntoineHX** [kornia/kornia#321]
Hi,

I think there's an error either in the implementation of kornia.filters.filter2D or in the documentation on the kernel shape.

```
k = torch.tensor([[[ 1.,  1.,  1.],
                   [ 1.,  5.,  1.],
                   [ 1.,  1.,  1.]]], device=device).expand(batch_size,-1,-1)
print(x.shape, k.shape)
smooth_x = kornia.filter2D(x, kernel=k, border_type='reflect', normalized=True) 
```

The Kernel should have the shape (B, kH, kW) but what's actually requested seems to be a shape (kH, kW). This sample code work only when i remove the expand().
This is probably because of this line in filter2D : `tmp_kernel = tmp_kernel.repeat(c, 1, 1, 1)`

**Comment by ucalyptus** [comment]
Faced the same error

**Comment by ducha-aiki** [comment]
Fixed in #327
